crates/pathfinder/src/bin/pathfinder/config.rs

```diff
@@ -55,6 +57,20 @@ Examples:
     )]
     rpc_address: SocketAddr,
 
+    #[arg(
+        long = "http-rpc.cors-domains",
+        long_help = r"Comma separated list of domains from which Cross-Origin requests will be accepted by the HTTP-RPC server.
+        
+Examples:
+    single: http://one.io
+    a list: http://first.com,http://second.com:1234
+    any:    *",
+        value_name = "DOMAIN-LIST",
+        value_delimiter = ',',
+        env = "PATHFINDER_HTTP_RPC_CORS_DOMAINS"
+    )]
+    rpc_cors_domains: Vec<String>,
```

Comment:
Does this not require a default value? Or does it default to empty? I'm always unsure with `clap` derive.

An option is instead of `"*"` to just have `empty` map to all allowed? Or what is the functional difference between `None` and `Some(AllowedOrigins::Any)`? This would probably simplify types and parsing a little bit.

Reply:
`clap` got me confused the first time I dived into a list of values, but it turns out that if:
- the command line option is missing you get an empty `vec![]`, which is the default value,
- the command line option is present but the value is empty, you also get the same `vec![]`

which in both cases means **cors disabled**. Going further, for cors enabled:
- any is `vec!["*"]`
- and some list is a non-empty vec that does not contain `"*"`

Ultimately this Vec<String> is an intermediate type only and not the config field we're using, because it is parsed ([here](https://github.com/eqlabs/pathfinder/blob/1fdb0a56be260808294afe7d7bf0868ebd3d748f/crates/pathfinder/src/bin/pathfinder/config.rs#L183) and [here](https://github.com/eqlabs/pathfinder/blob/1fdb0a56be260808294afe7d7bf0868ebd3d748f/crates/pathfinder/src/bin/pathfinder/config.rs#L340) into proper [`Option<AllowedOrigins>`](https://github.com/eqlabs/pathfinder/blob/1fdb0a56be260808294afe7d7bf0868ebd3d748f/crates/pathfinder/src/bin/pathfinder/config.rs#L258), where the meanings are:
- `None` for cors disabled
- `Some(AllowedOrigins::Any)` for any
- `Some(AllowedOrigins::List(vec))` for a vec of origins, where all of them are some specific valid origins **but not** a wildcard `"*"`

Reply:
What is the difference between cors disabled and `Any`?

Reply:
- `Some(AllowedOrigins::Any)` means that preflight request from any origin will be accepted
- `Some(AllowedOrigins::List)` means that preflight request from origins listed in the command line will be accepted
- None means that cors support is disabled and no preflight requests will be accepted

See [this test](https://github.com/eqlabs/pathfinder/blob/1fdb0a56be260808294afe7d7bf0868ebd3d748f/crates/rpc/src/middleware/cors.rs#L26)